examples/Multi_Armed_Bandit/bernoulli_multi_armed_bandit.py

```diff
@@ -0,0 +1,175 @@
+
+
+class SimulationResult(NamedTuple):
+    """Result of a bandit campaign simulation."""
+
+    earned_rewards: np.ndarray
+    estimated_win_rates: np.ndarray
+
+
+# Putting all pieces together, we define a function to simulate our campaign under a given acquisition strategy:
+
+
+def simulate(acqf: AcquisitionFunction) -> SimulationResult:
+    """Simulate the campaign with with the given acquisition function."""
+    recommender = TwoPhaseMetaRecommender(
+        initial_recommender=RandomRecommender(
+            allow_repeated_recommendations=True,
+            allow_recommending_already_measured=True,
+        ),
+        recommender=BotorchRecommender(
+            surrogate_model=surrogate,
+            acquisition_function=acqf,
+            allow_repeated_recommendations=True,
+            allow_recommending_already_measured=True,
+        ),
+    )
+
+    # Containers for storing the rewards and win rate estimates
+    earned_rewards = np.zeros((N_MC_RUNS, N_ITERATIONS))
+    estimated_win_rates = np.zeros((N_MC_RUNS, n_arms))
+
+    for mc in range(N_MC_RUNS):
+        campaign = Campaign(searchspace, objective, recommender)
+
+        for i in range(N_ITERATIONS):
+            df = campaign.recommend(batch_size=1)
+            recommended_arm_idx = int(df.iloc[0, 0].split("_")[1])
+            reward = BANDIT.draw_arm(recommended_arm_idx)
+            earned_rewards[mc, i] = reward
+            df["clicked"] = reward
+            campaign.add_measurements(df)
+
+        estimated_win_rates[mc] = surrogate.posterior_means()
+
+    return SimulationResult(earned_rewards, estimated_win_rates)
+
+
+### Running the Simulation
+
+# With our model in place, we can now simulate the campaign for both acquisition functions.
+# The success of our two objectives is quantified using the following metrics:
+# - For online optimization, we look at the **empirically achieved win rate** and
+#   compare it to the win rate of the best arm.
+# - For active learning, we report the **mean squared error of the win rates**
+#   estimated at the end of the campaign.
+
+for acqf in ACQFS:
+    name = acqf.__class__.__name__
+    result = simulate(acqf)
+    MSE = np.linalg.norm(result.estimated_win_rates - WIN_RATES, "fro") / n_arms
+
+    print(name)
+    print("-" * len(name))
+    print("Optimal win rate:", max(WIN_RATES))
+    print("Achieved win rate:", result.earned_rewards.mean())
+    print("Mean squared error (last iteration):", MSE)
+    print("\n")
+
+
+### Discussion
+
+# As expected, we observe a higher achieved win rate for the online optimization strategy, as its underlying acquisition function is designed to recommend the arm with the largest reward most often.
+# This comes at the cost of a higher estimation error for the remaining arms, which are sampled less frequently.
+# In contrast, the active learning strategy attempts to balance the estimation errors across arms, resulting in a significantly smaller overall error.
```

Comment:
I tested it locally and had a diffference of 0.07, I would not call that significantly...

Reply:
Calling 0.07 "insignificant" is meaningless without mentioning the relative scale. For me, I get something like:

<img width="472" alt="image" src="https://github.com/user-attachments/assets/9b6126cd-0e1f-42d9-91b1-2006e2cacc95">

This is like ~30% reduction in win rate when going to PSD with a similar reduction in MSE. How is this not significant?

Reply:
Ok, convinced.